docker/tarfile/dest.go

```diff
@@ -168,8 +172,10 @@ func (d *Destination) ReapplyBlob(ctx context.Context, info types.BlobInfo) (typ
 }
 
 func (d *Destination) createRepositoriesFile(rootLayerID string) error {
-	repositories := map[string]map[string]string{
-		d.reference.Name(): {d.reference.Tag(): rootLayerID}}
+	repositories := map[string]map[string]string{}
+	if d.reference != nil {
+		repositories[d.reference.Name()] = map[string]string{d.reference.Tag(): rootLayerID}
+	}
```

Comment:
Actually this should have been extended in #450 to handle `repoTags` as well.  Could you do that, please? And then `d.reference` can be completely removed in favor of `d.repoTags`.